lifx-core/src/lib.rs

```diff
@@ -67,6 +68,9 @@ pub enum Error {
 
     #[error("i/o error")]
     Io(#[from] io::Error),
+
+    #[error("raw message error:\n`{0}`")]
+    RawMessage(String),
```

Comment:
I think I eventually would like to make this `RawMessage(&'static str)` (so errors don't need to allocate a new String), but that doesn't need to happen in this pull request

Reply:
You won't be able to avoid an allocation though if you want to concatenate the error messages in the event multiple fields of the `Frame` are invalid; unless you're ok with just returning an error when the first invalid field is detected.

Reply:
I'm OK returning just the first error.   In practice that is what was happening with the asserts -- in the case of multiple invalid  fields, only the first assert would fire, so you'd have to fix-up that first issue before you knew about the others.

Reply:
Sounds good